app/post.go

```diff
@@ -49,3 +51,36 @@ func postHandler(c *gin.Context, app_settings common.AppSettings, database datab
 
 	return html_buffer.Bytes(), nil
 }
+
+func updatedPostHandler(c *gin.Context, app_settings common.AppSettings, database database.Database) ([]byte, error) {
```

Comment:
You don't need a new function here dude, just update the code for `postHandler` with the new checks you added.

I.e. check that the id is not negative in that function.